operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/DependentResource.java

```diff
@@ -5,8 +5,49 @@
 import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.javaoperatorsdk.operator.api.reconciler.Context;
 
+/**
+ * An interface to implement and provide dependent resource support.
+ *
+ * @param <R> the dependent resource type
+ * @param <P> the associated primary resource type
+ */
 public interface DependentResource<R, P extends HasMetadata> {
+
+  /**
+   * Reconciles the dependent resource given the desired primary state
+   *
+   * @param primary the primary resource for which we want to reconcile the dependent state
+   * @param context {@link Context} providing useful contextual information
+   * @return a {@link ReconcileResult} providing information about the reconciliation result
+   */
   ReconcileResult<R> reconcile(P primary, Context<P> context);
 
+  /**
+   * Retrieves the dependent resource associated with the specified primary one
+   *
+   * @param primaryResource the primary resource for which we want to retrieve the secondary
+   *        resource
+   * @return an {@link Optional} containing the secondary resource or {@link Optional#empty()} if it
+   *         doesn't exist
+   */
   Optional<R> getResource(P primaryResource);
+
+  /**
+   * Retrieves the resource type associated with this DependentResource
+   *
+   * @return the resource type associated with this DependentResource
+   */
+  Class<R> resourceType();
```

Comment:
Why we need this at this level?

Reply:
To make things explicit so that we don't need to rely on a default implementation that will not work in all situations.

Reply:
I mean are we sure that this will be needed in all cases? 
I mean I can imaging a dependent resource implementation that does not need resource type.

I thought it is something that is required for quarkus.

I guess we can then delete `ResourceTypeAware`.